workout-logger/scripts/test_gemini_api.py

```
@@ -0,0 +1,308 @@
+
+    # Unit Test: Retry parsing regex & RetryInfo extraction
+    sample_error = json.dumps({
+        "error": {
+            "code": 429,
+            "message": "Quota exceeded for metric: generativelanguage.googleapis.com/generate_content_free_tier_requests, limit: 20, model: gemini-3.6-flash. Please retry in 23.690750876s.",
+            "status": "RESOURCE_EXHAUSTED",
+            "details": [{"@type": "type.googleapis.com/google.rpc.RetryInfo", "retryDelay": "23.690750876s"}]
+        }
+    })
+    parsed_delay = extract_retry_delay(sample_error)
+    print(f"[TEST 1] Testing retryDelay parser on sample 429 payload:")
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)